frontend/ui/widget/toggleswitch.lua

```diff
@@ -94,15 +105,17 @@ end
 
 function ToggleSwitch:update()
     local pos = self.position
-    for i=1,#self.toggle_content do
-        if pos == i then
-            self.toggle_content[i].color = self.fgcolor
-            self.toggle_content[i].background = self.fgcolor
-            self.toggle_content[i][1][1].fgcolor = Blitbuffer.COLOR_WHITE
-        else
-            self.toggle_content[i].color = self.bgcolor
-            self.toggle_content[i].background = self.bgcolor
-            self.toggle_content[i][1][1].fgcolor = Blitbuffer.COLOR_BLACK
+    for i = 1, #self.toggle_content do
+        for j = 1, #self.toggle_content[i] do
+            if pos == (i - 1) * self.n_pos + j then
+                self.toggle_content[i][j].color = self.fgcolor
```

Comment:
This is not part of your change, so feel free to ignore my comment. But incase you don't mind to do the change, it's more readable to get a local reference of `self.toggle_content[i][j]` first and use it everywhere.